test/data.jl

```diff
@@ -61,3 +61,16 @@ end
     @test_throws ArgumentError load_german_credit(0)
     @test_throws ArgumentError load_german_credit(-100)
 end
+
+@testset "UCI Adult dataset" begin
+    counterfactual_data = load_uci_adult()
+    @test size(counterfactual_data.X)[2] == 1000
+    @test size(counterfactual_data.X)[1] == 14
+    @test size(counterfactual_data.y)[2] == 1000
+
+    counterfactual_data = load_uci_adult(500)
+    @test size(counterfactual_data.X)[2] == 500
+    @test size(counterfactual_data.X)[1] == 14
+    @test size(counterfactual_data.y)[2] == 500
+    @test_throws ArgumentError load_uci_adult(0)
```

Comment:
You could also add a test here that checks for an error when n is negative. Though this wouldn't increase code coverage, it would make the code more mutation-safe.

Reply:
Also, if you add a check that `n` is not bigger than the size of the dataset, then it would be good to add a test which checks that the error is indeed thrown when `n` is excessively big.

Reply:
Good point! I've added a check for n > 32000 (size of the dataset being 32561).